### PR TITLE
SDK-306: Integrated Micro Frontend Tooling Setup in OpenMRS SDK - Refactoring

### PR DESCRIPTION
**Pull request by wikumChamith**
Rename Setup.setConfig to Setup.createConfigurationFolder and Setup.downloadConfig to Setup.downloadConfiguration

Jira ticket: https://issues.openmrs.org/browse/SDK-306

### CODE REVIEW

**Comment by wikumChamith** [comment]
> @wikumChamith have you tried to run the `mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:4.6.0-SNAPSHOT:setup` command outside of the sdk source folder?

No. Is it possible to run this from anywhere?? We haven't made a release too.

**Comment by wikumChamith** [comment]
> @wikumChamith have you tried to run the `mvn org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:4.6.0-SNAPSHOT:setup` command outside of the sdk source folder?

@dkayiwa I just tried this. It is not showing all the server versions.

`[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- openmrs-sdk-maven-plugin:4.6.0-SNAPSHOT:setup (default-cli) @ standalone-pom ---

Setting up a new server...

Specify server id (-DserverId) (default: 'server15'): 

You can setup the following servers:
1) 2.x Distribution
2) Platform
3) 03 Distribution

Which one do you choose? [1/2/3]: 3

You can deploy the following versions of O3:
1) Reference Application 3.0.0-SNAPSHOT
2) Other...

`

**Comment by wikumChamith** [comment]
@dkayiwa Is it possible to run this from anywhere?? I thought we have to make a release for this.

**Comment by dkayiwa** [comment]
@wikumChamith i made a release for 4.6.0
Can you try it out?

**Comment by wikumChamith** [comment]
> @wikumChamith i made a release for 4.6.0 Can you try it out?

Just checked it, Still got the error. Any idea what is causing it??

**Comment by dkayiwa** [comment]
Which error?

**Comment by wikumChamith** [comment]
> Which error?

SDK only showing Reference Application 3.0.0-SNAPSHOT.  I am getting this error after selecting it.

`Unable to find/resolve artifact. Failure to find org.openmrs.module:initializer-omod:jar:2.5.0 in https://mavenrepo.openmrs.org/public was cached in the local repository, resolution will not be reattempted until the update interval of openmrs-repo has elapsed or updates are forced `

**Comment by wikumChamith** [comment]
@ibacher I made the change you requested.

**Comment by dkayiwa** [comment]
> SDK only shows Reference Application 3.0.0-SNAPSHOT.

Can you add this to your settings.xml file? https://pastebin.com/jMrC2XJQ

**Comment by wikumChamith** [comment]
> > SDK only shows Reference Application 3.0.0-SNAPSHOT.
> 
> Can you add this to your settings.xml file? https://pastebin.com/jMrC2XJQ

Now It is working fine. Will other users get the same error??

**Comment by dkayiwa** [comment]
> Will other users get the same error??

Yes they will get the same error for as long as they do not have that repository in their settings.xml file

**Comment by wikumChamith** [comment]
> > Will other users get the same error??
> 
> Yes they will get the same error for as long as they do not have that repository in their settings.xml file

What should we do about it??

**Comment by dkayiwa** [comment]
> What should we do about it??

Update the documentation to reflect it.

**Comment by wikumChamith** [comment]
> > What should we do about it??
> 
> Update the documentation to reflect it.

Does this impact only users who already have the SDK, or does it also apply to new users?

**Comment by dkayiwa** [comment]
> Does this impact only users who already have the SDK, or does it also apply to new users?

It applies to whoever does not have that repository in their settings.xml file

**Comment by wikumChamith** [comment]
> > What should we do about it??
> 
> Update the documentation to reflect it.

@dkayiwa I updated the documentation. https://wiki.openmrs.org/pages/viewpage.action?pageId=224527013

**Comment by wikumChamith** [comment]
@dkayiwa, @ibacher  is everything ok with this??

**Comment by dkayiwa** [comment]
@wikumChamith can you create a talk post and ask frontend devs to try set up an O3 backend server using the SDK?

**Comment by dkayiwa** [comment]
> I updated the documentation. https://wiki.openmrs.org/pages/viewpage.action?pageId=224527013

Awesome! 👍

**Comment by wikumChamith** [comment]
> @wikumChamith can you create a talk post and ask frontend devs to try set up an O3 backend server using the SDK?

@dkayiwa I created a talk post: https://talk.openmrs.org/t/deploying-o3-with-the-sdk/39925

File: sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java [hunk 1]
```diff
@@ -779,7 +779,7 @@ private Map<String, String> getDistroVersionsOptionsMap(Set<String> versions, Ve
 	private Map<String, String> getO3VersionsOptionsMap(VersionsHelper versionsHelper,
 			String optionTemplate, String artifactTemplate) {
 		Map<String, String> optionsMap = new LinkedHashMap<>();
-		Artifact artifact = new Artifact("referenceapplication-distro", "3.0.0-SNAPSHOT", "org.openmrs.distro", "zip");
+		Artifact artifact = new Artifact("referenceapplication-distro", "version", "org.openmrs.distro", "zip");
```
Comment (ibacher) [inline sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java:782]: Does this work with the string `"version"`? I would've thought this should be `null` or something?

**Comment by wikumChamith** [comment]
new PR: https://github.com/openmrs/openmrs-sdk/pull/224

**Pull request closed without merging**